September Activity Report (#425)
haiku/website: content/blog/bitigchi/2020-10-01-haiku_monthly_activity_report_september_2020.md

+++
type = "blog"
title = "Haiku Activity Report - September 2020"
author = "bitigchi"
date = "2020-10/01 13:00:00+03:00"
tags = ["monthly activity report"]
+++

Welcome to the Haiku Monthly Activity Report for September 2020. This report covers revisions from hrev54539 to hrev54608.

## System Sounds Contest

> We've started a contest for the system sounds, and currently it's ongoing. For more details, see the contest post.

Author: Oops.

PulkoMandy fixed various issues with MediaPlayer and ffmpeg implementation that caused seeking and cover art display problems.

Murai Takashi made some code cleanups in various system parts.

syedsouban and bitigchi implemented custom search engine shortcuts in WebPositive. Currently following shortcuts are supported:

- Google (g)
- Bing (b)
- Wikipedia (w)
- DuckDuckGo (d)
- Baidu (a)
- Yandex (y)
- Ecosia (e)
- Qwant (q)

## Compatibility

CodeForEvolution implemented the `qsort_r` function.

## Documentation

John Scipione improved the documentation for BMessage and AppKit.

nielx implemented layout customisations for the Doxygen stylesheet.

## File Systems

brjhaiku implemented the initial methods required for btrfs write support.

## Kernel

Jerome Duval implemented support for more CPU power modes and functions that allows system to use the CPU power more efficiently. This fixed an issue where on some hardware the system clock speed was stuck at 1.5 GHz, and improved performance on other systems.

## Package Management

> Andrew Lindesay fixed a crash in HaikuDepot whilst synchronising package metadata.

Reviewer: I think https://git.haiku-os.org/haiku/commit/?id=hrev54558 should also be mentioned here.

## Preferences

> Preetpal Kaur's GSOC project work has been merged, and she continues to work on final polishes and bug fixes.

Reviewer 2: ```suggestion
Preetpal Kaur's GSOC project work has been merged, and she continues to work on final polishing and bug fixes.
```

## POSIX

rofl0r has fixed the prototype of gethostbyaddr to be more in-line with FreeBSD and Linux.

## RISC-V

kallisti5 added u-boot support for RISC-V.

## System Kits

John Scipione fixed a regression that prevented character drawing when a tab is inserted.

Pascal Abresch implemented enabling adding custom user TLS certificates.

> PulkoMandy implemented creating catalog files from external source files.

Reviewer 3: Ok, so what I did there is allow loading a catalog without needing an executable object (application, library or add-on) for it.
This is used to localize static libraries and would also allow localizing, for example, bash or python apps (or any other interpreted language) using the locale kit APIs.

## System Servers

> PulkoMandy fixed a regression that caused unpersistant brightness levels across different workspaces. This eventually led fixing a system crash as well.

Reviewer 2: ```suggestion
PulkoMandy fixed a regression that reset brightness levels to 100% on all but the first workspace.
```
That crash was introduced with the brightness-level fix, I think.

## System Translators

Gerasim Troeglazov fixed a crash that occurs while displaying `.icns` files.

------

If you would like to try out these changes right away, grab a copy of the latest [nightlies](https://download.haiku-os.org) and give it a go!

> Haiku Project always welcomes new faces. If you are a developer, sysadmin, designer, or a writer, we will always appreciate your support. Come meet us at the [Freenode #haiku IRC channel](irc://chat.freenode.net/haiku),  [forums](https://discuss.haiku-os.org) or introduce yourself in the [Haiku mailing list](https://www.freelists.org/list/haiku).

Reviewer 3: I'd say "new hands" rather than "new faces"?

Author: It's a nice pun, but sounds too meta. As if we only welcome the work done by hands(?!) rather than the person him/herself.

Reviewer 3: well that's how it would be in French, I think. But maybe it doesn't work too well in English then.

> Haiku Project always welcomes new faces. If you are a developer, sysadmin, designer, or a writer, we will always appreciate your support. Come meet us at the [Freenode #haiku IRC channel](irc://chat.freenode.net/haiku),  [forums](https://discuss.haiku-os.org) or introduce yourself in the [Haiku mailing list](https://www.freelists.org/list/haiku).

Reviewer 2: ```suggestion
The Haiku Project always welcomes new faces. If you are a developer, sysadmin, designer, or a writer, we'd appreciate your support. Come meet us at the [Freenode #haiku IRC channel](irc://chat.freenode.net/haiku),  [forums](https://discuss.haiku-os.org) or introduce yourself on the [Haiku mailing list](https://www.freelists.org/list/haiku).
```
